pkg/cmd/pr/create/create.go

```diff
@@ -273,7 +281,7 @@ func createRun(opts *CreateOptions) (err error) {
 	if err != nil && !errors.As(err, &notFound) {
 		return fmt.Errorf("error checking for existing pull request: %w", err)
 	}
-	if err == nil {
+	if err == nil && !opts.DryRun {
```

Comment:
Can you explain why this is desirable behaviour? I assume it's to aid in testing, but I'm not totally clear on the workflow you have that makes this useful.

Reply:
See https://github.com/cli/cli/pull/8376/commits/efe5a9022e2f42dcb86241b35d0b175e450af5eb